src/parser/mod.rs

```diff
@@ -3188,19 +3196,24 @@ impl<'a> Parser<'a> {
     pub fn expected<T>(&self, expected: &str, found: TokenWithLocation) -> Result<T, ParserError> {
         parser_err!(
             format!("Expected: {expected}, found: {found}"),
-            found.location
+            found.span.start
         )
     }
 
     /// If the current token is the `expected` keyword, consume it and returns
     /// true. Otherwise, no tokens are consumed and returns false.
     #[must_use]
-    pub fn parse_keyword(&mut self, expected: Keyword) -> bool {
+    pub fn parse_keyword(&mut self, expected: Keyword) -> Option<TokenWithLocation> {
```

Comment:
Wouldn't it be easier (and less breaking) to keep the existing `parse_keyword` function that returns a bool and add a new `get_keyword` or `parse_keyword_token` or `parse_keyword_with_loc` that returns a token with location ?

Reply:
That's fine with me, I just figured no reason to have two functions that do the same thing, but one just omits information. And going forward I expect more and more places in the parser to want the Token version (error handling as well as more source spans). I don't expect any downstream users to be using internal parser methods, but I might be wrong. 😅 Hyrum's Law and all

Reply:
I reverted the change and added `parse_keyword_token` as via your suggestion. I also changed `parse_keyword` to be implemented as `parse_keyword_token(keyword).is_some()` to not duplicate code. 😃